lib/cxx-api/include/clingo/control.hh

```diff
@@ -349,6 +304,82 @@ class Control {
         ground(PartSpan{parts}, std::move(ctx));
     }
 
+    //! Ground the logic program with the given parameters.
+    //!
+    //! The given parts determine which program parts are grounded with which
+    //! paramaters. If no parts are given, the parts given by the parts
+    //! directive are grounded.
+    //!
+    //! A handler implementing the GroundEventHandler interface can be given to
+    //! handle grounding events. The handler can be passed by
+    //! value/reference/pointer. If passed by value, the handler must be
+    //! copyable. If passed by reference/pointer, lifetime must be managed by
+    //! the caller.
+    //!
+    //! @param parts the parts to ground the control object with
+    //! @param handler the context to use for grounding
+    template <Detail::UserData<GroundEventHandler> Handler = std::nullptr_t>
+    [[nodiscard]] auto start_ground(std::optional<PartSpan> parts = std::nullopt, Handler &&handler = nullptr) const
+        -> GroundHandle {
+        std::optional<PartVector> default_parts;
+        std::vector<clingo_part_t> c_parts;
+        if (default_parts = !parts ? this->parts() : std::nullopt; default_parts) {
+            parts = default_parts;
+        }
+        if (parts) {
+            c_parts.reserve(parts->size());
+            for (auto const &part : *parts) {
+                c_parts.emplace_back(part.name.data(), part.name.size(), c_cast(part.params.data()),
+                                     part.params.size());
+            }
+        } else {
+            c_parts.reserve(1);
+            c_parts.emplace_back("base", 4, nullptr, 0);
+        }
```

Comment:
Consider extracting this into its own function so that it can be called from here and `ground()` above.